### PR TITLE
Dowshift.js case study

### PR DESCRIPTION
**Pull request by essmahr**
A blog post outlining some of the maybe-interesting stuff we did when rewriting our `Select` to use Downshift.js.

- fixed a spacing issue with the code blocks so that they space the same as paragraphs
- added hr styling to posts so that we can separate content with `---`

### CODE REVIEW

File: _posts/2021-04-18-a-complex-select-refactor-with-downshift-js/index.md [hunk 1]
```diff
@@ -0,0 +1,722 @@
...
+
+We like Downshift because:
+
+1. **It's headless.** It packages no UI of its own, giving our excellent
+   designers free reign.
+2. **It's accessibility-first,** fully aligns with the aria
+   [combobox](https://www.w3.org/TR/wai-aria-practices/examples/combobox/aria1.1pattern/listbox-combo.html)
+   and
+   [listbox](https://www.w3.org/TR/wai-aria-practices/examples/listbox/listbox-collapsible.html)
+   specs. Not only is this great for accissibility, it opens our UI up to the
```
Comment (AlecRosenbaum) [inline _posts/2021-04-18-a-complex-select-refactor-with-downshift-js/index.md:113]: > accissibility

Comment (essmahr) [reply _posts/2021-04-18-a-complex-select-refactor-with-downshift-js/index.md]: fixed!

File: _posts/2021-04-18-a-complex-select-refactor-with-downshift-js/index.md [hunk 2]
```diff
@@ -0,0 +1,722 @@
+---
+layout: post
+title: 'Highlights From a Complex Downshift.js Refactor'
+date: 2021-04-18
```
Comment (AlecRosenbaum) [inline _posts/2021-04-18-a-complex-select-refactor-with-downshift-js/index.md:4]: also might want to update the date

Comment (essmahr) [reply _posts/2021-04-18-a-complex-select-refactor-with-downshift-js/index.md:4]: definitely -- will do so at publish time

**Review by lukasmlady — APPROVED** [review]
Looks great Scott, thanks for writing this up!

I've left a couple of inline comments, mostly typos.

🎉 🚀

File: _posts/2021-04-18-a-complex-select-refactor-with-downshift-js/index.md [hunk 3]
```diff
@@ -0,0 +1,728 @@
...
+---
+
+It should come as no suprise that, as part of the
+[app.close.com](https://app.close.com) UI codebase, we on the Frontend team
+maintain a set of reusable React components. It's 2021, after all! It seems
+everyone and their dog has a design system now.
+
+As it goes with any component library, our components range from the exceedingly
+simple (looking at you, `<LineSeparator/>`) to tremendously complex. And the
+component that wins the for sheer lines of code is by far and away our
```
[the remark this answers is not in the document]
Comment (essmahr) [reply _posts/2021-04-18-a-complex-select-refactor-with-downshift-js/index.md]: missing word, fixed

**Pull request merged**